File: 2024/06/api/Models/MyData.Model.cs
public abstract class MyDataBase
{
    public abstract int Code { get; set; }
    public abstract string Name { get; set; }
}

public class MyData : MyDataBase
{
    public override int Code { get; set; }
    public override string Name { get; set; }
}


public class NewMyData: MyDataBase
{
    public override int Code { get; set; }
    public override string Name { get; set; }
    public string NewData { get; set; } = "";
}

File: 2024/06/front/src/app/models/my-data.model.ts
export abstract class MyDayaBase {
  abstract code: number;
  abstract name: string;
}

export interface MyData extends MyDayaBase {
  code: number;
  name: string;
}

export interface NewMyData extends MyDayaBase {
  code: number;
  name: string;
  newData: string;
}